server/src/main/java/org/elasticsearch/action/search/SearchPhaseExecutionException.java

```diff
@@ -85,8 +85,9 @@ private static Throwable deduplicateCause(Throwable cause, ShardSearchFailure[]
     @Override
     public RestStatus status() {
         if (shardFailures.length == 0) {
-            // if no successful shards, it means no active shards, so just return SERVICE_UNAVAILABLE
-            return RestStatus.SERVICE_UNAVAILABLE;
+            // if no successful shards, the failure could have happened even before the fetch phase started
+            // so try to get the status from the cause instead of returning SERVICE_UNAVAILABLE blindly
+            return ExceptionsHelper.status(this.getCause());
```

Comment:
the only doubt I have about this change is that we may end up returning 500 instead of 503, for instance whenever the cause is null, and probably in other cases too. Should we at least keep 503 in case the cause is null?

Reply:
Leaving null to 500 was intentional. Ideally speaking, being not able to parse the exception correctly is server side (elasticsearch here) error, and should be 500. Receiving 500 on client side indicates potential bug and helps us iterate faster to address other scenarios.

Reply:
I get what you are saying but here we are returning 503 for a specific case, where there are no shared failures yet we got an error. I am not sure we should go from 503 to 500 in that case, but I will discuss it with the team.

Reply:
Do you have any updates on this?

Reply:
not yet, thanks for the ping!

Reply:
we discussed with @javanna and we think that we can preserve the old behavior if the cause is null. In such case we can continue to return a `503`, this is important for actions like https://github.com/elastic/elasticsearch/blob/9e706966282eaadafc77727059dd4194cbed492a/server/src/main/java/org/elasticsearch/action/search/InitialSearchPhase.java#L152 that implies that setting shard failures to an empty array will return the right status.
So can you change this to:
````
return getCause() == null ? RestStatus.SERVICE_UNAVAILABLE : ExceptionsHelper.status(this.getCause());
````

Reply:
@jimczi Thanks for highlighting this use case. I have updated the PR with proposed changes.